Odata-docs/webapi-8/tutorials/using-utf8jsonwriter-to-improve-serialization-performance.md

```diff
@@ -0,0 +1,34 @@
+---
+title:  "Using Utf8JsonWriter to improve serialization performance"
+description: Learn how to configure ASP.NET Core OData 8 to use Utf8JsonWriter to improve serialization performance. 
+date:   2023-08-11
+ms.date: 08/11/2023
+author: habbes
+ms.author: clhabins
+---
+
+# Using Utf8JsonWriter to improve serialization performance
+
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+`Microsoft.OData.Core` version 7.12.2 introduced a new JSON writer that’s based on .NET’s [`Utf8JsonWriter`](/dotnet/api/system.text.json.utf8jsonwriter). This writer offers better performance than the JSON writer that is used by default in `Microsoft.OData.Core`. To learn more, [visit this page](../../odatalib/v7/using-utf8jsonwriter-for-better-performance.md). In this article, we are going to show you how to configure your ASP.NET Core OData application to use the `Utf8JsonWriter`.
+
+This feature is available in ASP.NET Core OData version 8.0.11 or later.
```

Comment:
I think this is confusing.. Which feature is availabe here? Is it the use of utf8jsonwriter? Somewhere up there you indicated that people can enable utf8jsonwriter in `Microsoft.AspNetCore.OData 8 and 7`

Reply:
I think you should make it clear which version of ODL has this change..Then all versions of Microsoft.AspNetCore.OData with that ODL version have the feature.

Reply:
The feature is available in both 8.x and 7.x. But in 8.x it's only available in 8.0.11 or later, i.e. not available in 8.0.10 earlier. This particular article exists in the WebAPI 8 docs page. There are separate articles in the 7.x docs which specify the minimum version of 7.x supported: https://learn.microsoft.com/en-us/odata/webapi/using-utf8jsonwriter-to-improve-serialization-performance.

The article already mentions which version of ODL introduces this feature: 7.12.2. I think it's also relevant to mention the version of WebAPI 8 that introduces this feature. Someone may be using an older version like 8.0.7. I think it's easier to tell them this is feature requires 8.0.11 or later than to ask them to go through all the 8.x versions to and look for the ones that depend on ODL 7.12.2 or later.

Reply:
@habbes I know it'd involve a bit more work, but would it be better to capture the supported versions here:
![image](https://github.com/MicrosoftDocs/OData-docs/assets/28291332/d0f9f32c-723a-4cb7-bc2f-d12832aa0969)
That is add includes specific to the version(s) that support `ODataUtf8JsonWriter`. That way the doc doesn't say at the top that the feature is supported in the V8 and then in the body it says 8.0.11.

Reply:
@habbes Another suggestion would be to rephrase the paragraph to immediately establish the connection between Microsoft.OData.Core 7.12.2 and Microsoft.AspNetCore.OData 8.0.11, e.g.:

> The [Microsoft.AspNetCore.OData](https://www.nuget.org/packages/Microsoft.AspNetCore.OData) library takes a dependency on the [Microsoft.OData.Core](https://www.nuget.org/packages/Microsoft.OData.Core) library. [Microsoft.OData.Core 7.12.2](https://www.nuget.org/packages/Microsoft.OData.Core/7.12.2) introduced a new JSON writer that's based on .NET's [Utf8JsonWriter](/dotnet/api/system.text.json.utf8jsonwriter). This new writer offers better performance than the JSON writer that is used by default. Reference [Microsoft.AspNetCore.OData 8.0.11](https://www.nuget.org/packages/Microsoft.AspNetCore.OData/8.0.11) or higher to use the new writer. To learn more, [visit this page](../../odatalib/v7/using-utf8jsonwriter-for-better-performance.md).

> In this article, we are going to show you how to configure your ASP.NET Core OData application to use the `Utf8JsonWriter`.

Then you don't need the sentence: "This feature is available in ASP.NET Core OData version 8.0.11 or later."

Reply:
> @habbes I know it'd involve a bit more work, but would it be better to capture the supported versions here: ![image](https://private-user-images.githubusercontent.com/28291332/310039566-d0f9f32c-723a-4cb7-bc2f-d12832aa0969.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk2OTk0NDcsIm5iZiI6MTcwOTY5OTE0NywicGF0aCI6Ii8yODI5MTMzMi8zMTAwMzk1NjYtZDBmOWYzMmMtNzIzYS00Y2I3LWJjMmYtZDEyODMyYWEwOTY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA0MjU0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0MDkxNjcxYTdiNWFlMzVlY2ZmZGVlYjg0MDZlMmE5ZWRmYmUwYWUwNjgxOTFiMmNiYjE4M2YyODI4MDdhNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.OmNLJGSRLQGFdExvUeTMc6uEL88gpG4tIxqM3bylDYc) That is add includes specific to the version(s) that support `ODataUtf8JsonWriter`. That way the doc doesn't say at the top that the feature is supported in the V8 and then in the body it says 8.0.11.

@gathogojr do you mean adding an includes specific for 8.0.11 and later? Or? I'm not sure I understood the statement. Note that this particular article sits in the Web API 8 docs tree, which is separate from the Web API 7 docs tree.

Reply:
@gathogojr I've rephrased the statement as suggestion, except that I didn't mention that part about "The Microsoft.AspNetCore.OData library takes a dependency on Microsoft.OData.Core library". Not sure if that's really necessary. But open to adding it if you think otherwise.